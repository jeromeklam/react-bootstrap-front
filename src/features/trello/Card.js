import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import Tag from './Tag';
import CardMenu from './CardMenu';

import { Row, Col } from '../grid';

class Card extends Component {
  onDelete = e => {
    this.props.onDelete();
    e.stopPropagation();
  };

  render() {
    const {
      showDeleteButton,
      style,
      tagStyle,
      onClick,
      className,
      id,
      title,
      label,
      description,
      tags,
      cardDraggable,
      t
    } = this.props;

    return (
      <article data-id={id} onClick={onClick} style={style} className={classnames('trello-card', className)}>
        <Row>
          <Col size={20}>
            <span className="trello-card-title text-secondary" draggable={cardDraggable}>
              {title}
            </span>
          </Col>
          <Col size={10} textAlign="right">
            <span className="trello-card-right">{label}</span>
          </Col>
          <Col size={6} textAlign="right">
            <CardMenu delete={showDeleteButton} onUpdate={onClick} onDelete={this.onDelete} t={t}/>
          </Col>
        </Row>
        <div className="trello-card-detail">{description}</div>
        {tags && tags.length > 0 && (
          <div className="trello-card-footer">
            {tags.map(tag => (
              <Tag key={tag.title} {...tag} tagStyle={tagStyle} />
            ))}
          </div>
        )}
      </article>
    );
  }
}

Card.propTypes = {
  showDeleteButton: PropTypes.bool,
  onDelete: PropTypes.func,
  onClick: PropTypes.func,
  style: PropTypes.object,
  tagStyle: PropTypes.object,
  className: PropTypes.string,
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  label: PropTypes.string,
  description: PropTypes.string,
  tags: PropTypes.array,
};

Card.defaultProps = {
  showDeleteButton: true,
  onDelete: () => {},
  onClick: () => {},
  style: {},
  tagStyle: {},
  title: 'no title',
  description: '',
  label: '',
  tags: [],
  className: '',
};

export default Card;
