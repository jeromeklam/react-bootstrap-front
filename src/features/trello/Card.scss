.trello-card {
  border-radius: 3px;
  border-bottom: 1px solid #ccc;
  background-color: #fff;
  position: relative;
  padding: 10px;
  cursor: pointer;
  max-width: 350px;
  margin-bottom: 7px;
  min-width: 330px;

  &:hover {
    background-color: theme-color('primary-light');
    color: #000;
  }
}

.trello-card-header {
  margin-bottom: 10px;
  display: flex;
  flex-direction: row;
  align-items: flex-start;
  border-bottom: 1px solid #eee;
  padding-bottom: 6px;
  color: #000;
}

.trello-card-title {
  font-weight: normal;
  line-height: 18px;
  font-size: 14px;
  word-break: break-all;
  white-space: nowrap;
  overflow: hidden;
}

.trello-card-project {
  font-weight: normal;
  line-height: 18px;
  font-size: 12px;
  word-break: break-all;
  white-space: nowrap;
  overflow: hidden;
}

.trello-card-detail {
  border-top: 1px solid #eee;
  padding-top: 3px;
  margin-top: 3px;
  font-size: 12px;
  color: #4d4d4d;
  white-space: pre-wrap;
  line-height: 1;
}

.trello-card-footer {
  border-top: 1px solid #eee;
  padding-top: 3px;
  margin-top: 3px;
  text-align: right;
}

.trello-card-priority {
  width: 18px !important;
  height: 18px !important;
  text-align: center;
  line-height: 16px !important;
  font-size: 12px;
  border-radius: 9px !important;
  &.border-simple {
    border: 1px solid !important;
  }
  &.border-warning {
    border: 2px solid !important;
  }
  &.border-danger {
    border: 2px solid !important;
  }
  margin-top: 4px;
}

.trello-card-deadline {
  text-align: left;
  font-size: 11px;
  margin-top: 3px;
}

.trello-card-progress {
  text-align: left;
  font-size: 11px;
  margin-top: 3px;
}

.trello-card-comment {
  width: 30%;
  text-align: center;
  font-size: 12px;
}
