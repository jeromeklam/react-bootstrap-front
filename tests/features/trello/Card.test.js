import React from 'react';
import renderer from 'react-test-renderer';
import Card from '../../../src/features/trello/Card';

const callback = jest.fn();

it('render with className and name', () => {
  // First render
  const component = renderer.create(
    <Card className="btn bg-secondary" onClick={callback} t={callback}>
      My Button
    </Card>
  );
  let test = component.toJSON();
  expect(test).toMatchSnapshot();
});
