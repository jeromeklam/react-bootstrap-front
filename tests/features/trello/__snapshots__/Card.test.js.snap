// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render with className and name 1`] = `
<article
  className="trello-card btn bg-secondary"
  onClick={[MockFunction]}
  style={Object {}}
>
  <div
    className="trello-card-header"
  >
    <div
      className="trello-card-title text-secondary"
    >
      no title
    </div>
    <span
      className="trello-card-right"
    >
      
    </span>
    <div
      className="trello-delete-button"
      onClick={[Function]}
    >
      <button
        className="btn btn-xs text-danger"
      >
        âœ–
      </button>
    </div>
  </div>
  <div
    className="trello-card-detail"
  >
    
  </div>
</article>
`;
